services/reproducible-central/reproducible-central.service.js

```diff
@@ -0,0 +1,93 @@
+import Joi from 'joi'
+import { BaseJsonService, pathParams } from '../index.js'
+
+const schema = Joi.object().pattern(Joi.string(), Joi.string()).required()
```

Comment:
I'd like us to be more specific about the validation here please.

It seems like the possible values we are expecting to accept here are:

- N digits, a literal '/' character, and then N more digits
- A literal 'X' character
- A literal '?' character

Given that, I'd suggest we make the schema

```js
const schema = Joi.object()
  .pattern(Joi.string(), Joi.string().regex(/^\d+\/\d+$|^[X?]$/))
  .required()
```

Then we know we're not going to be dealing with anything unexpected.

Reply:
thanks for the code: I was just not able to write it myself :)